save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -26,11 +23,7 @@ internal class ContestController(
      */
     @GetMapping("/{contestName}")
     @PreAuthorize("permitAll()")
-    fun getContestByName(@PathVariable contestName: String) = Mono.fromCallable {
-        contestService.findByName(contestName)
-    }.switchIfEmpty {
-        Mono.error(ResponseStatusException(HttpStatus.NOT_FOUND))
-    }
+    fun getContestByName(@PathVariable contestName: String) = justOrNotFound(contestService.findByName(contestName))
```

Comment:
These endpoints are intended to be consumed by external clients, including our frontend. So they shouldn't return Java classes in Jackson's format, instead we should create and use `ContestDto` here

Reply:
Done